chat-plugins/mafia.js

```diff
@@ -977,6 +978,52 @@ exports.commands = {
 		},
 		unlynchhelp: [`/mafia unlynch - Withdraw your lynch vote. Fails if your not voting to lynch anyone`],
 
+		'!selflynch': true,
+		enableself: 'selflynch',
+		selflynch: function (target, room, user, connection, cmd) {
+			let targetRoom = room;
+			target = target.split(',');
+			if (Rooms(target[0]) && Rooms(target[0]).users[user.userid]) targetRoom = Rooms(target.shift());
+			if (!targetRoom || !targetRoom.game || targetRoom.game.gameid !== 'mafia') return this.errorReply(`There is no game of mafia running in this room.`);
+			if (!user.can('mute', null, room) && targetRoom.game.hostid !== user.userid) return user.sendTo(targetRoom, `|error|/mafia ${cmd} - Access denied.`);
+			let action = toId(target.shift()), game = targetRoom.game;
+			if (!action) return this.parse(`/help mafia selflynch`);
+			if (this.meansYes(action)) {
+				if (game.selfEnabled === 'hammer') {
+					game.sendRoom(`Selfhammering has been changed to Selflynching.`, {declare: true});
+				} else if (!game.selfEnabled) {
+					game.sendRoom(`Selflynching has been enabled.`, {declare: true});
+				} else {
+					return user.sendTo(targetRoom, `|error|Selflynching is already set.`);
+				}
+				game.selfEnabled = true;
+				game.updateHost();
```

Comment:
You should update all players since the lynch buttons avaliable will change. (do this for the other two as well)